### PR TITLE
chore: fix build for upstream-fef2098

### PR DESCRIPTION
**Pull request by Karrq**
# What :computer: 
* Fixes build for `fef2098`

### CODE REVIEW

**Review by elfedy — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ env:
 
 jobs:
     doctest:
-        runs-on: ubuntu-22.04-github-hosted-16core
+        runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply .github/workflows/test.yml:23]: I believe where it's being done like that they are quite agnostic to the ubuntu version, but for release and test workflows it's better to keep them versioned.

File: crates/script/src/simulate.rs [hunk 2]
```diff
@@ -112,10 +112,11 @@ impl PreSimulationState {
         // Executes all transactions from the different forks concurrently.
         let futs = transactions
             .into_iter()
-            .map(|mut transaction| async {
+            // NOTE(zk): this one is `mut` upstream but doesn't need to be
```
Comment (nbaztec) [inline crates/script/src/simulate.rs:115]: Remove this comment now? (and one below)

File: crates/zksync/core/src/convert.rs [hunk 3]
```diff
@@ -150,17 +150,23 @@ impl ConvertSignature for ZkSignature {
...
+            self.s.to_b256(),
+            parity.y_parity(),
+        )
     }
 }
 
 impl ConvertSignature for AlloySignature {
     fn to_ethers(self) -> ZkSignature {
-        let v = self.v();
+        let parity = alloy_primitives::Parity::from(self.v());
```
Comment (nbaztec) [inline crates/zksync/core/src/convert.rs:164]: Shouldn't this already be of the type `Parity`?

File: crates/zksync/core/src/convert/eip712.rs [hunk 4]
```diff
@@ -133,6 +133,27 @@ impl Transaction for Eip712SignableTransaction {
...
+            } else {
+                // otherwise return the max fee per gas
+                self.inner.max_fee_per_gas.low_u128()
+            }
+        })
+    }
+
+    fn is_dynamic_fee(&self) -> bool {
+        true
+    }
```
Comment (nbaztec) [inline crates/zksync/core/src/convert/eip712.rs:156]: Would be deleted code when you merge `main`

File: crates/zksync/core/src/convert.rs (hunk 3 above)
Comment (Karrq) [reply crates/zksync/core/src/convert.rs:164]: unfortunately in the `PrimitiveSignature` it's just a bool

File: crates/zksync/core/src/convert/eip712.rs (hunk 4 above)
Comment (Karrq) [reply crates/zksync/core/src/convert/eip712.rs:156]: I know but still need it to make this compile for now

File: .github/workflows/test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Karrq) [reply .github/workflows/test.yml:23]: clippy for example is using ubuntu-latest

[the remark this answers is not in the document]
Comment (nbaztec) [reply .github/workflows/test.yml:23]: Clippy isn't affected by system-related behavior

File: crates/script/src/simulate.rs (hunk 2 above)
Comment (nbaztec) [reply crates/script/src/simulate.rs:115]: I don't get it, it seems to be inconsistent with the code

**Review by nbaztec — APPROVED** [review]

**Pull request merged**